### PR TITLE
Add InfernalVoidScans

### PR DESCRIPTION
**Pull request by niclimcy**

### CODE REVIEW

**Comment by niclimcy** [comment]
Couldn't make a source request on your discord so I did it myself

**Comment by TheNetsky** [comment]
Have you tested this?

File: src/InfernalVoidScans/InfernalVoidScans.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+    //----HOMESCREEN SELECTORS
+    //Disabling some of these will cause some Home-Page tests to fail, edit these test files to match the setting.
+    //Always be sure to test this in the app!
+
+    override homescreen_PopularToday_enabled = true
+
+    override homescreen_LatestUpdate_enabled = true
+
+    override homescreen_NewManga_enabled = false
```
Comment (TheNetsky) [inline src/InfernalVoidScans/InfernalVoidScans.ts]: You sure about this?

File: src/InfernalVoidScans/InfernalVoidScans.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    override homescreen_TopAllTime_enabled = true
```
Comment (TheNetsky) [inline src/InfernalVoidScans/InfernalVoidScans.ts]: I don't see the widget on the homepage?

File: src/InfernalVoidScans/InfernalVoidScans.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    override homescreen_TopMonthly_enabled = true
```
Comment (TheNetsky) [inline src/InfernalVoidScans/InfernalVoidScans.ts]: see above

File: src/InfernalVoidScans/InfernalVoidScans.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    override homescreen_TopWeekly_enabled = true
```
Comment (TheNetsky) [inline src/InfernalVoidScans/InfernalVoidScans.ts]: see above

**Review by TheNetsky — COMMENTED** [review]
Are you sure you've tested this?

**Comment by niclimcy** [comment]
Not sure if you've seen the recommendations widget before with genre selectors

**Comment by TheNetsky** [comment]
> Not sure if you've seen the recommendations widget before with genre selectors

This is not related to that recommendation widget, since that one isn't used.
For the month/day/alltime to be enabled it needs this widget
image

Also I'm pretty sure homescreen_NewManga_enabled  can be abled since the homepage has this section:
image

I hope you've tested the rest of the changes if they actually work within the app?

**Comment by niclimcy** [comment]
Testing right now, I'll push something to add the last widget since I didn't see it initially

**Comment by niclimcy** [comment]
Is homescreen_NewManga_selector case sensitive?

**Comment by niclimcy** [comment]
I can see all the widgets now

**Comment by TheNetsky** [comment]
> I can see all the widgets now

Genre tags don't seem to be working, edit the selectors or see if you need to use the other prebuild selectors method

**Comment by niclimcy** [comment]
> > I can see all the widgets now
> 
> Genre tags don't seem to be working, edit the selectors or see if you need to use the other prebuild selectors method

Which other source has those genre tags? I want to reference off them

**Comment by TheNetsky** [comment]
> > > I can see all the widgets now
> > 
> > 
> > Genre tags don't seem to be working, edit the selectors or see if you need to use the other prebuild selectors method
> 
> Which other source has those genre tags? I want to reference off them

Unsure, I've given 2 templates within every source which are the most common one. Usually if one doesn't work, the other one will. Else you need to edit the selectors. I'd say just look at the other sites included in MangaStream.

**Comment by niclimcy** [comment]
Actually it seems like all the mangastream sources do not have Genres working, I checked a few like "ImperfectComics", "LelManga", "ReadKomik"

Current genre tag is correct

**Comment by TheNetsky** [comment]
> Actually it seems like all the mangastream sources do not have Genres working, I checked a few like "ImperfectComics", "LelManga", "ReadKomik"
> 
> Current genre tag is correct

Not "all"
Some indeed never worked
https://lelmanga.com/ never worked since they aren't displayed anywhere (nor does it have a /genres path)
same goes for https://readkomik.com/ (this one is experimental since the theme is very much different than the usual ones)

https://imperfectcomic.com/ is indeed broken, they updated their theme. Which does seem to be the same version as the site you're trying to add. See if you can play around with some selectors to make it work.

**Comment by niclimcy** [comment]
Alright I updated imperfect comic's tag too, works

**Comment by niclimcy** [comment]
My discord is nicknitewolf#2790 btw

**Comment by TheNetsky** [comment]
Perfect!
Good catch and thanks for the help!

**Pull request merged**